src/functionalTest/java/uk/gov/hmcts/dts/fact/admin/AdminCourtSpoeAreaOfLawEndpointTest.java

```diff
@@ -0,0 +1,195 @@
+            SPOE_AREA_OF_LAW_ENDPOINT,
+            Map.of(AUTHORIZATION, BEARER + forbiddenToken)
+        );
+        assertThat(response.statusCode()).isEqualTo(FORBIDDEN.value());
+    }
+
+    /************************************************************* Spoe Get Request Tests for a court. ***************************************************************/
+
+    @Test
+    public void returnSpoeAreasOfLawForTheCourt() {
+        final var response = doGetRequest(
+            ABERYSTWYTH_JUSTICE_CENTRE_SPOE_AREAS_OF_LAW_PATH,
+            Map.of(AUTHORIZATION, BEARER + superAdminToken)
+        );
+        assertThat(response.statusCode()).isEqualTo(OK.value());
+
+        final List<AreaOfLaw> areasOfLaw = response.body().jsonPath().getList(".", AreaOfLaw.class);
+        assertThat(areasOfLaw).isNotEmpty();
+    }
+
+    @Test
+    public void shouldRequireATokenWhenGettingSpoeAreasOfLawForTheCourt() {
+        final var response = doGetRequest(ABERYSTWYTH_JUSTICE_CENTRE_SPOE_AREAS_OF_LAW_PATH);
+        assertThat(response.statusCode()).isEqualTo(UNAUTHORIZED.value());
+    }
+
+    @Test
+    public void shouldBeForbiddenForGettingSpoeAreasOfLawForTheCourt() {
+        final var response = doGetRequest(
+            ABERYSTWYTH_JUSTICE_CENTRE_SPOE_AREAS_OF_LAW_PATH,
+            Map.of(AUTHORIZATION, BEARER + forbiddenToken)
+        );
+        assertThat(response.statusCode()).isEqualTo(FORBIDDEN.value());
+    }
+
+    @Test
+    public void shouldReturnNotFoundWhenSpoeCourtDoesNotExist() {
+        final Response response = doGetRequest(
+            ABERYSTWYTH_COURT_AREAS_OF_LAW_NOT_FOUND_PATH,
+            Map.of(AUTHORIZATION, BEARER + superAdminToken)
+        );
+        assertThat(response.statusCode()).isEqualTo(NOT_FOUND.value());
+    }
+    /************************************************************* PUT request tests section. ***************************************************************/
+
+    @Test
+    public void shouldUpdateCourtSpoeAreaOfLaw() throws JsonProcessingException {
+        final List<AreaOfLaw> currentCourtSpoeAreaOfLaw = getCurrentSpoeAreaOflaw();
+        final List<AreaOfLaw> expectedCourtAreasOfLaw = updateCourtAreasOfLaw(currentCourtSpoeAreaOfLaw);
+        final String updatedJson = objectMapper().writeValueAsString(expectedCourtAreasOfLaw);
+        final String originalJson = objectMapper().writeValueAsString(currentCourtSpoeAreaOfLaw);
+        final var response = doPutRequest(
+            ABERYSTWYTH_JUSTICE_CENTRE_SPOE_AREAS_OF_LAW_PATH,
+            Map.of(AUTHORIZATION, BEARER + superAdminToken),
+            updatedJson
+        );
+        assertThat(response.statusCode()).isEqualTo(OK.value());
+
+        final List<AreaOfLaw> updatedCourtAreaOfLaw = response.body().jsonPath().getList(
+            ".",
+            AreaOfLaw.class
+        );
+        assertThat(updatedCourtAreaOfLaw.get(0).getName()).isEqualTo(expectedCourtAreasOfLaw.get(0).getName());
```

Comment:
You may want to check that the returned list in its entirety, is the same as the one you expect (rather than just one element of it). I.e

assertThat(updatedCourtAreaOflaw, is(expectedCourtAreasOfLaw)); - which will check that all the elements and values match. 

I could be getting the wrong end of the stick here though!

Reply:
actually, the returned list populate all values to null except ID, Name and SPOE. so the returned entire list can't be same as expected, thats why i only put assertions on ID, Name.